scanapi/evaluators/code_evaluator.py

```diff
@@ -27,6 +31,11 @@ def evaluate(cls, sequence, vars, is_a_test_case=False):
         code = match.group("python_code")
         response = vars.get("response")
 
+        if not response and cls.response_var_pattern.search(code):
+            if is_a_test_case:
+                return (True, None)
```

Comment:
Maybe the test should fail, because we are not asserting the statement at all. Also, I believe that for both cases, when it is and when it is not a test_case, we should raise a warning. 

What do you think?

Reply:
Sorry @camilamaia but I don't understand when you say raise a warning.

Reply:
No problem, I was suggesting to print something in the console to warn the developer that the line was skipped because there is no response. Otherwise, the developer might not notice that they are writing something wrong in their specification.

The code would be something like:

```python
import logging
...

logger = logging.getLogger(__name__)
...

def method():
    ...
    logger.warning(message)
```

Here is an example with logger.error:

https://github.com/scanapi/scanapi/blob/3e09a669dae83d26c397d0c63804f464eb34f73a/scanapi/scan.py#L29

The only difference would be changing error for warning.

https://docs.python.org/3/library/logging.html#logging.warning

Does it make sense now?

Reply:
Yes! I now understand, but same times it is not a issue. I think this warning could be shown when response is skipped in body, headers, url and params, not in vars.

What do you think?

Reply:
Hmmm, would you mind giving me an example using `vars` where it is not an issue? Just because probably I am not seeing a case that you are.

The way I am seeing it, if the user is trying to access the response, but response is not there yet, it should be an issue, we are just skipping it. Am I missing a scenario?

An example where it is an issue:

```yaml
endpoints:
  name: my_endpoint
  path: foo
  vars: 
    id: ${{ response.json()['id'] }}
  requests:
    - ...
```

Reply:
On your example, the response is not an issue, because vars is evaluate before and after request.

https://github.com/scanapi/scanapi/blob/3e09a669dae83d26c397d0c63804f464eb34f73a/scanapi/tree/request_node.py#L131-L149

But the example below, name in body, access the response out of scope:

```
endpoints:
  name: my_endpoint
  path: foo
  vars: 
    id: ${{ response.json()['id'] }}
  requests:
    body:
      name: ${{ response.json()['name'] }}
    - ...
```

The last pair programming, we decide create a filter for response to apply before of the request, but later on, I found a issue in update order that can be see below, that after resolved the response now can be skipped in code_evaluator.py. So I didn't created the response filter.

https://github.com/scanapi/scanapi/blob/3e09a669dae83d26c397d0c63804f464eb34f73a/scanapi/evaluators/spec_evaluator.py#L26-L27

Confuse!? :grimacing:

Reply:
Ohhh, now I got it! Thanks for the detailed explanation. So let me check if I got it right:

```
`response` variable can only be accessed inside the `vars` and `tests` scopes.
```

Maybe the warning message can mention this sentence ⬆️

Reply:
Thinking better, I will do what we define in pair programming. Because, filtering the response var in vars scope before request I can raise error in code evaluate instead skip it.

Reply:
That works for me. I like the idea of raising the error and not masking it (only skipping it). 👍